Python/Product/PythonTools/PythonTools/Repl/PythonCommonInteractiveEvaluator.cs

```diff
@@ -184,7 +184,13 @@ internal abstract class PythonCommonInteractiveEvaluator :
         }
 
         public virtual Uri DocumentUri { get => _documentUri; protected set => _documentUri = value; }
-        public virtual Uri NextDocumentUri() => new Uri(DocumentUri, $"#{++_nextDocumentIndex}");
+        public virtual Uri NextDocumentUri() {
+            var d = DocumentUri;
+            if (d != null) {
+                return new Uri(d, $"#{++_nextDocumentIndex}");
+            }
+            return null;
```

Comment:
I'm trying to find out what will happen when this returns `null`.

`PythonTextBufferInfo.DocumentUri` would return `null`.

That's used in `IntellisenseController.AnalyzerBufferAsync` and passed to `ProjectAnalyzer.AnalyzeFileAsync`, which will throw ANE,

It will be caught with `HandleAllException` in `IntellisenseController..ConnectSubjectBuffer` but that's going to log the error?

Probably not the right thing to do since debug repl is not an unexpected/exceptional case? We should avoid calling into `AnalyzeFileAsync` in the first place?

Reply:
Probably. If we don't have a document URI then we really shouldn't be analyzing anything. The tricky part is making sure that it doesn't get cached and never updated when you start debugging. I'll take a closer look this afternoon.